mltsp/featurize.py

```diff
@@ -9,301 +5,55 @@
-                            'target': target_value}
-                    elif len(line.split(',')) > 2:
-                        els = line.strip().split(',')
-                        fname, target_value = els[:2]
-                        other_metadata = els[2:]
-                        # Convert to floats, if applicable:
-                        for i in range(len(other_metadata)):
-                            try:
-                                other_metadata[i] = float(
-                                    other_metadata[i])
-                            except ValueError:
-                                pass
-                        fname_target_dict[fname] = target_value
-                        fname_target_science_features_dict[fname] = {
-                            'target': target_value}
-                        fname_metadata_dict[fname] = dict(
-                            list(zip(other_metadata_labels,
-                                     other_metadata)))
-            line_no += 1
-    return (features_to_use, fname_target_dict,
-            fname_target_science_features_dict,
-            fname_metadata_dict)
-
-
-def generate_featurize_input_params_list(features_to_use, fname_target_dict,
-                                         fname_target_science_features_dict,
-                                         fname_metadata_dict, zipfile_path,
-                                         custom_script_path, is_test):
-    """
-    """
-    input_params_list = []
-
-    zipfile = tarfile.open(zipfile_path)
-    unzip_dir = tempfile.mkdtemp()
-    zipfile.extractall(path=unzip_dir)
-    all_fnames = zipfile.getnames()
-    num_objs = len(fname_target_dict)
-
-    for fname in all_fnames:
-        if is_test and len(input_params_list) >= 3:
-            break
-        short_fname = shorten_fname(fname)
-        if os.path.isfile(fname):
-            ts_path = fname
-        elif os.path.isfile(os.path.join(unzip_dir, fname)):
-            ts_path = os.path.join(unzip_dir, fname)
+from .celery_tasks import featurize_ts_file as featurize_celery_task
+from . import featurize_tools as ft
+
+
+def write_features_to_disk(featureset, featureset_id,
+                             in_docker_container=False):
+    featureset_dir = "/tmp" if in_docker_container else cfg.FEATURES_FOLDER
+    featureset_path = os.path.join(featureset_dir,
+                                   "{}_featureset.nc".format(featureset_id))
+    featureset.to_netcdf(featureset_path)
+
+
+def load_and_store_feature_data(features_path, featureset_id="unknown",
+                                in_docker_container=False, first_N=None):
+    targets, metadata = ft.parse_headerfile(features_path)
+    if first_N:
+        metadata = metadata[:first_N]
+        if targets is not None:
+            targets = targets[:first_N]
+    featureset = ft.assemble_featureset([], targets, metadata)
+    write_features_to_disk(featureset, featureset_id, in_docker_container)
```

Comment:
I think `in_docker_container` is obsolete here now - everywhere except in `custom_feature_tools`? I've been meaning to pull it out everywhere it's no longer used.